### PR TITLE
Return all results if User Sync History report if filter is empty

### PR DESCRIPTION
**Pull request by NoahCarnahan**
Previously, now results were shown if the "Select Mobile Worker" filter was empty and displaying the help text "All Mobile Workers". Now it shows data for all mobile workers.
http://manage.dimagi.com/default.asp?166564

FYI code review buddy @biyeun

### CODE REVIEW

File: corehq/apps/reports/standard/deployments.py [hunk 1]
```diff
@@ -165,18 +165,20 @@ def rows(self):
 
         user_id = self.request.GET.get('individual')
         if not user_id:
-            return []
-
-        # security check
-        get_document_or_404(CommCareUser, self.domain, user_id)
-
+            key_args = {}
```
Comment (czue) [inline corehq/apps/reports/standard/deployments.py]: Am I missing something or is this not applying any project-level filtering?

Comment (czue) [inline corehq/apps/reports/standard/deployments.py]: Assuming that's right - and that we don't have a couch view we can use, I think the easiest thing is to just force people to select a user.

Comment (czue) [inline corehq/apps/reports/standard/deployments.py]: (I'm also happy to take this if you want since it's sort of a QA bug on my feature)

**Comment by czue** [comment]
put don't merge on because i think as-is this is a security loophole

File: corehq/apps/reports/standard/deployments.py (hunk 1 above)
Comment (NoahCarnahan) [inline corehq/apps/reports/standard/deployments.py]: Ah ha. Yeah, this definitely won't work. I don't see another view for `SyncLog`s. Easiest thing I think would be to update the placeholder text to say "Select a user" or something like that.

Comment (czue) [inline corehq/apps/reports/standard/deployments.py]: that sounds like a good interim plan to me

Comment (czue) [inline corehq/apps/reports/standard/deployments.py]: other options would be:
- use the domain/docs view somehow
- grab all the user IDs in the domain first and then query by those

I don't think either of those will be particularly performant or easy to order though, so happy to just force people to pick.

**Pull request closed without merging**